Ft artisan request response 130385627

morriskimani opened the pull request:
#[130385627] Artisans should be able to accept or reject service requests

#### What does this PR do?
This PR implements adds functionality that allows an artisan, upon receiving a notification that a task has been assigned to him, to accept or reject the task. Should an artisan accept a quote, then he should be able to give a quote(price) and submit that quote. 
This PR also adds functionality for marking service request notifications as expired.
Finally, this PR adds functionality for notifying a tasker when his task has been accepted by an artisan provided by workdey.

#### Screenshots (if appropriate)
- The service request
<img width="1280" alt="notification_request" src="https://cloud.githubusercontent.com/assets/20963766/19875472/09afbdb4-9fde-11e6-9bed-4916dc0c0371.png">

- On clicking accept, a quote field appears
<img width="1280" alt="notification_request_enter_quote" src="https://cloud.githubusercontent.com/assets/20963766/19875484/221e47a8-9fde-11e6-91e4-7f17d67a8807.png">

- Task is marked as accepted when artisan submits quote
<img width="1280" alt="task_accepted_with_quote" src="https://cloud.githubusercontent.com/assets/20963766/19875547/a804c144-9fde-11e6-97e8-33e21078fe94.png">

- Task is marked as rejected when artisan rejects it
<img width="1280" alt="task_rejected" src="https://cloud.githubusercontent.com/assets/20963766/19875526/656fb550-9fde-11e6-90b0-198ef20e0d46.png">

Commit:
test(Quotes): add tests for the quote model

Commit:
chore: format code to match style guide

Commit:
test: isolate failing test

Commit:
chore: update seeds

Commit:
chore: change configurations for circle-ci so that migrations can run before deployment

Commit:
chore: remove commented out code

Commit:
fix: correct ruby version to be used in circle-ci

Commit:
feat(Quote): replace alert boxes with plain text when showing error messages

Commit:
feat(Quote): add styling for quote error messages

Commit:
fix(Quote): fix issue where a service request notification was being marked as read even before the artisan accepts or rejects the request

Commit:
fix: fix error where a an artisan can enter non-numeric characters

Commit:
Merge branch 'master' into ft-artisan-request-response-130385627

Commit:
bg: fix failing test

Commit:
chore: configure RSpec to run in documentation mode

hmasila commented on app/controllers/notifications_controller.rb:
```diff
@@ -30,12 +30,12 @@ def update
       if @reply_to_sender == true
         @notification.reply_to_sender(@message, @event_name)
-        @notification.update_as_read
+        @notification.update_as_read force_update: true
```
Why force the update?

morriskimani replied:
I need that to solve a bug where a  notification was being marked as `read` even before the artisan takes action i.e. accepts or rejects a quote

hmasila commented on spec/features/share_contact_spec.rb:
```diff
@@ -37,32 +37,19 @@
-     end
+         expect(page).to have_content(artisan.firstname)
+         expect(page).to have_content(artisan.email)
+       end
```
check indentation

morriskimani replied:
done

hmasila commented on spec/features/share_contact_spec.rb:
```diff
@@ -72,8 +59,15 @@ def share_contact(share = true)
+    quote = create(:quote, artisan_id: artisan.id)
+    create(
+      :notification,
+      receiver_id:tasker.id,
```
Put the space after colon

Commit:
chore: format code to match style guide

Commit:
fix(Notifications): fix bug where expired notifications were not being marked as read

Commit:
fix(Notifications): fix bug where Service notifications were not being marked as read after a user has seen and acted on the notification

Commit:
feat(Quotes): Allow user to enter non numeric characters when giving quote

This is a better approach than restricting the characters that a user can type. When non-numeric characters cannot be entered, user experience may be hampered since a user might think that the application isn't responding.

Commit:
chore: remove the reply_to_sender functionality after a service has been accepted or rejected
